### PR TITLE
Machine Learning Regression Evaluation Metrics

### LINKED ISSUE #4365: (Machine Learning)Machine Learning Regression Evaluation Metrics
#### Proposed title of article
Machine learning regression evaluation metrics

#### Proposed article introduction
We can determine the performance of a regression model  by  the error rate of the predictions the model makes. We can also measure the performance by knowing how well the regression line fit the dataset. Choosing a proper metric is crucial while evaluating regression models.

 In this article we are going to discuss the regression evaluation metrics. we are also going to discuss the advantages and disadvantages of each one of them. Finally we will discuss what a good regression model is.

#### Key takeaways 

- Regression evaluation metrics.
- Advantages and disadvantages of each discussed metric.
- Understand what a good regression model is.

#### Article quality
The available articles explain these metrics in a shallow way. Some discuss only the popular metrics. This article will be more in-depth and will be covering all the metrics.

#### References
N/A

### PR DESCRIPTION
**Pull request by FINCH285**
@Bayler 
This closes #4365

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Plagiarism check is done. Please work on the highlighted issues in the attached pdf. Thanks
[New Scan 5_40 PM Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7506888/New.Scan.5_40.PM.Copyleaks.Report.pdf)

**Comment by FINCH285** [comment]
@Bayler
 Finished working on the highlighted issues.

File: content/articles/machine-learning-regression-evaluation-metrics/index.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+**Disadvantages**
+- We have to apply Gradient descent to the graph of *MAE* to make it differentiable.
+
+### Mean Squared Error (MSE)
+This metric is almost similar to *MAE*. The difference is that in *MSE* instead of using absolute values, we use the squares of the difference between the predicted values and the actual values. In the case of data outliers, *MSE* will be larger than *MAE* since we are squaring the error.
+
+ *MSE* is calculated using the following formula.
+
+![MSE](/engineering-education/machine-learning-regression-evaluation-metrics/mse.png)
+
```
Comment (Bayler) [inline content/articles/machine-learning-regression-evaluation-metrics/index.md:73]: Did you forget to upload the file mse.png? I can't find it. In addition, I can also see a file named log.png but yet it hasn't been used in your article. Please sort that out.

File: content/articles/machine-learning-regression-evaluation-metrics/index.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+*n* is the total number of data points.
+
+We calculate *MAE* by following the steps below:
+1. Calculate the residual (error) of each data point.
+2. Calculate the absolute value. This is to get rid of the sign.
+3. Calculate the average of all residuals.
+
+If *MAE* is zero, the model predictions are perfect. The higher the *MAE* the worse the predictions are.
+
```
Comment (Bayler) [inline content/articles/machine-learning-regression-evaluation-metrics/index.md:59]: Perfect? Really? I don't think model predictions can be perfect. Please use the correct word.

File: content/articles/machine-learning-regression-evaluation-metrics/index.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+- [Adjusted R Squared](#adjusted-r-squared)
+- [Root Mean Squared Log Error(*RMSLE*)](#root-mean-squared-log-error)
+- [What is a good regression model](#what-is-a-good-regression-model)
+- [Conclusion](#Conclusion)
+
+### Introduction
+There are numerous regression evaluation metrics. All these metrics aim at showing us the prediction error of our model. An error is defined as the difference between the value the model predicts and the actual value. The lower the error, the better the performance of the model, and the higher the error, the worse the performance of the model.
+
+![Reference](/engineering-education/machine-learning-regression-evaluation-metrics/reference.jpeg)
+
```
Comment (Bayler) [inline content/articles/machine-learning-regression-evaluation-metrics/index.md]: I'm assuming this images aren't yours. If so, please add the sources. Please do this for all your images

File: content/articles/machine-learning-regression-evaluation-metrics/index.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+We calculate *RMSE* by following the steps below:
+
+1. Calculate the error for each data point.
+2. Calculate the squared value of the residuals.
+3. Calculate the average of the squared residuals.
+4. Obtain the square root of the answer in step 3.
+
+**Advantages**
+- When it comes to outliers, *RMSE* is more sensitive than *MAE*.
```
Comment (Bayler) [inline content/articles/machine-learning-regression-evaluation-metrics/index.md:101]: Sensitive to what exactly? Let it be clear

File: content/articles/machine-learning-regression-evaluation-metrics/index.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+**Disadvantages**
+- When compared to *MAE* it is less robust to outliers.
+
+### R Squared (R²)
+This metric is used to describe the goodness of fit of linear regression. We also refer to it as the coefficient of determination. Simply put R² is used to show : 
+
+"How the change in the output (y) is a result of the change in the input (x).
+
+We use the following formula to calculate *RMSE*:
```
Comment (Bayler) [inline content/articles/machine-learning-regression-evaluation-metrics/index.md:110]: RMSE? Did you mean R Squared?

File: content/articles/machine-learning-regression-evaluation-metrics/index.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+- It cannot determine whether the predictions are biased.
+
+### Adjusted R Squared
+We use Adjusted R-squared to figure out how reliable the correlation is. We determine this by adding new independent variables. It shows whether the added predictors improve the regression model or they are of no value.
+
+We use the following formula to calculate *adjusted r squared*:
+
+![Adjusted](/engineering-education/machine-learning-regression-evaluation-metrics/adjusted.png)
+
+Adjusted R-squared will penalize us for adding independent variables that do not fit the model. The independent variable is represented by *k* in the above formula.
```
Comment (Bayler) [inline content/articles/machine-learning-regression-evaluation-metrics/index.md:133]: I cannot see the independent variable `k` in the formula

**Comment by Bayler** [comment]
@FINCH285 Hi. Please work on the suggested corrections before we can move on for final review.

**Comment by Bayler** [comment]
@FINCH285 Please refer to this [article](https://github.com/section-engineering-education/engineering-education/pull/4566/files) on line 52 to learn how to reference an image source that's not yours

**Comment by Bayler** [comment]
@FINCH285 Please take note of all the commits I've made so that you can better your next article.

**Comment by FINCH285** [comment]
@Bayler Thanks for the review. I'll take note of all the commits

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
👍 overall nice work - very detailed and straight forward tutorial 
 
Please review and suggestions/changes made by your peer reviewer and/or myself during a final review.

One thing to note (and we can fix it afterwards for this article) but when including formulas its always best to include the actual formula (via text) so that the reader (audience) can copy it if needed instead of switching back and forth to write down what is in the image. @FINCH285 

We can add a mathjax script that renders all the formulas. 

cc @Bayler 

Overall nice and easy to follow along. 👍

**Pull request merged**